File: src/main/java/org/fountainmc/api/event/server/ServerStopEvent.java
package org.fountainmc.api.event.server;

import javax.annotation.Nonnull;

import org.fountainmc.api.Server;
import org.fountainmc.api.event.Event;

import static com.google.common.base.Preconditions.checkNotNull;

public interface ServerStopEvent extends Event {

    @Nonnull
    public Server getServer();

    public static ServerStopEvent create(Server server) {
        checkNotNull(server, "Null server");
        return () -> server;
    }

}

File: src/main/kotlin/org/fountainmc/api/event/server/ServerStopEvent.kt
package org.fountainmc.api.event.server

import org.fountainmc.api.Server
import org.fountainmc.api.event.Event

interface ServerStopEvent : Event {

    val server: Server

    companion object {
        fun create(server: Server): ServerStopEvent {
            class SimpleServerStopEvent(override val server: Server): ServerStopEvent
            return SimpleServerStopEvent(server)
        }
    }
}
